table-api/src/main/java/io/deephaven/api/TableOperations.java

```diff
@@ -1300,4 +1300,31 @@ TOPS updateBy(UpdateByControl control, Collection<? extends UpdateByOperation> o
     TOPS ungroup(boolean nullFill, Collection<? extends ColumnName> columnsToUngroup);
 
     // -------------------------------------------------------------------------------------------
+
+    /**
+     * Drop the {@code columnNames} from {@code this} table.
+     *
+     * @param columnNames the columns to drop
+     * @return the table
+     */
+    TOPS dropColumns(String... columnNames);
+
+    /**
+     * Drop the {@code columnNames} from {@code this} table.
+     *
+     * @param columnNames the columns to drop
+     * @return the table
+     */
+    TOPS dropColumns(Collection<String> columnNames);
+
+    /**
+     * Drop the {@code columnNames} from {@code this} table.
+     *
+     * @param columnNames the columns to drop
+     * @return the table
+     */
+    TOPS dropColumns(ColumnName... columnNames);
```

Comment:
You're missing `@ConcurrentMethod` annotations.
Also, it seems to me like "this" isn't the right way to say what dropColumns does. We return a new table without the dropped columns.

Reply:
Will fix. Made a change to docs, happy to take more specific wording.